src/lib/logUtils.js

```diff
@@ -0,0 +1,63 @@
+import { CLIENT_VERSION } from './util';
+
+import Browser from './Browser';
+
+/**
+ * Get current time in ISO format.
+ *
+ * @return {string} The time in ISO format.
+ */
+export function getISOTime() {
+    return new Date().toISOString();
+}
+
+/**
+ * Generates a GUID/UUID compliant with RFC4122 version 4.
+ *
+ * @return {string} A 36 character uuid
+ */
+export function uuidv4() {
+    /* eslint-disable */
+    function generateRandom(c) {
+        const r = (Math.random() * 16) | 0;
+        const v = c == 'x' ? r : (r & 0x3) | 0x8;
+        return v.toString(16);
+    }
+    /* eslint-enable */
+
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, generateRandom);
+}
+
+// Unique ID of this Preview session. Needs to be generated AFTER function def.
+const PREVIEW_SESSION_ID = uuidv4();
```

Comment:
Is this safe? What if the file gets cached? We're never explicitly importing PREVIEW_SESSION_ID so when does it get executed?

Reply:
This'll only be created when the file is parsed and executed by the browser, meaning that cache will have no affect on it.